PiBox.Hosting/WebHost/src/PiBox.Hosting.WebHost/Services/TypeImplementationResolver.cs

```diff
@@ -36,6 +38,36 @@ private object GetArgument(Type instanceType, Type type)
                 if (type.HasAttribute<ConfigurationAttribute>())
                     return GetConfiguration(type, type.GetAttribute<ConfigurationAttribute>()!.Section);
 
+                var isList = type.GetInterfaces().Any(i => i.IsAssignableTo(typeof(IEnumerable)));
+                var innerType = !isList ? type : type.GetElementType() ?? type.GenericTypeArguments[0]!;
+
+                if (innerType.IsInterface && innerType.IsAssignableTo(typeof(IPluginConfigurator)))
+                {
+                    if (!isList)
+                        throw new NotSupportedException(
+                            $"For plugin configurators you must implement to take in an enumerable of the configurators. Configurator: {innerType.Name}");
+                    var args = _resolvedTypes.Select(x => x.Assembly).Distinct()
+                        .SelectMany(x => x.GetTypes())
+                        .Where(x => x.IsClass && !x.IsAbstract && x.IsAssignableTo(innerType))
+                        .Select(ResolveInstance)
+                        .ToList();
+                    if (!isList)
```

Comment:
code is unreachable as you're throwing an appropriate Exception above already, remove/redo